chap-model.tex

```diff
@@ -1087,17 +1086,45 @@ \subsection{Failure Modes and Exceptions}
 exceptions may be delivered, and there is flexibility in the CHERI model in
 when exceptions might be delivered.
 For example, while an attempt to broaden (rather than narrow) bounds could
-generate an immediate exception (our prototyping choice), the operation could
-instead generate a non-dereferenceable pointer as its output, in effect
-deferring an exception until the time of an attempted load, store, or
-instruction fetch.
-The former offers slightly improved debuggability (by exposing the error
-earlier), whereas the latter can offer microarchitectural benefits by
-reducing the set of instructions that can throw exceptions.
+generate an immediate exception, the operation could instead generate a
+non-dereferenceable pointer as its output, in effect deferring an exception
+until the time of an attempted load, store, or instruction fetch.
 Both of these implementations ensure monotonicity by preventing derived
 pointers from improperly allowing increased access following guarded
 manipulation, and are consistent with the model.
 
+Initially, in our prototyping, we selected to deliver exceptions as early as
+possible when such events occur.
+However, all current CHERI ISA instantiations defer exceptions to the use
+of a capability's authority, instead clearing the tag on operations that
+would otherwise violate monotonicity.
+
+\pdrnote{The following two paragraphs perhaps belong in rationale?}
+The early exception approach offers slightly improved debuggability
+(by exposing the error earlier).
+However, early exceptions limit compiler optimization as instructions that may
+throw exceptions are restricted in how they can safely be reordered.
+For example, a bounds restriction performed within a loop could be hoisted to
+before the loop, except for the possibility that the bounds are being widened.
```

Comment:
Maybe something more detailed like this:

```suggestion
For example, this prevents a bounds restrictions performed within a loop from
being hoisted outside the loop, unless that instructions is always executed.
If the loop is not always entered, this could turn a conditional execution of a  trapping instruction into an unconditional one.\footnote{%
This is not just a theoretical possibility -- we observed this happening in the FreeBSD kernel and had to modify the compiler to avoid hoisting any
potentially-trapping CHERI instructions.
}
```